spyder/plugins/console/widgets/main_widget.py

```diff
@@ -0,0 +1,535 @@
+            ConsoleWidgetActions.ToggleWrap,
+            text=_("Wrap lines"),
+            toggled=lambda val: self.set_option('wrap', val),
+            initial=self.get_option('wrap'),
+        )
+        codecompletion_action = self.create_action(
+            ConsoleWidgetActions.ToggleCodeCompletion,
+            text=_("Automatic code completion"),
+            toggled=lambda val: self.set_option('codecompletion/auto', val),
+            initial=self.get_option('codecompletion/auto'),
+        )
+
+        # Submenu
+        internal_settings_menu = self.create_menu(
+            ConsoleWidgetMenus.InternalSettings,
+            _('Internal console settings'),
+        )
+        self.add_item_to_menu(
+            buffer_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            wrap_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            codecompletion_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+        self.add_item_to_menu(
+            exteditor_action,
+            menu=internal_settings_menu,
+            section=ConsoleWidgetInternalSettingsSubMenuSections.Main,
+        )
+
+        # Options menu
+        options_menu = self.get_options_menu()
+        self.add_item_to_menu(
+            run_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            environ_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            syspath_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            internal_settings_menu,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Run,
+        )
+        self.add_item_to_menu(
+            quit_action,
+            menu=options_menu,
+            section=ConsoleWidgetOptionsMenuSections.Quit,
+        )
+
+        self.shell.set_external_editor(
+            self.get_option('external_editor/path'), '')
+        self.start_interpreter(self.get_option('namespace'))
+
+    def on_option_update(self, option, value):
+        if option == 'max_line_count':
+            self.shell.setMaximumBlockCount(value)
+        elif option == 'wrap':
+            self.shell.toggle_wrap_mode(value)
+        elif option == 'codecompletion/auto':
+            self.shell.set_codecompletion_auto(value)
+        elif option == 'external_editor/path':
+            self.shell.set_external_editor(value, '')
+
+    def update_actions(self):
+        pass

```

Comment:
Why this `pass` here? Please add a docstring to this method describing why that's necessary.

Reply:
This method is a required part of the PluginMainWidget API, but it does not need to do anything on this case